File: DoctorService/Entities/Doctor.cs
namespace DoctorService.Entities
{
    public class Doctor : BaseEntity
    {
        public string? UserId { get; set; }
        public string FullName { get; set; }
        public string Identification { get; set; }
        public string SpecialityId { get; set; }
        public Speciality Speciality { get; set; }
        public User? User { get; set; }
    }
}

File: web-app/src/app/core/models/doctor.ts
import { Base } from './base';
import { Speciality } from './speciality';

export interface Doctor extends Base {
  fullName: string;
  specialityId: string;
  identification: string;
  speciality?: Speciality;
}

export interface DoctorRecord extends Base {
  fullName: string;
  specialityId: string;
  identification: string;
  speciality?: string;
}

export interface DoctorCreateDto {
  fullName: string;
  specialityId: string;
  identification: string;
}
